### PR TITLE
Add VRRPv3 support for IPv6 address list (#2430)

### LINKED ISSUE #2430: Add support for IPv6 addresses in VRRPv3 layer
#### Brief description

The VRRPv3 layer has a field addrlist which supports IPv4 addresses only.

This prevents use of the layer for generating IPv6 VRRPv3 advertisement packets.

#### Environment

-   Scapy version: `v2.4.3-242-gdcfb63d0`
-   Python version: `3.7.6`
-   Operating System: `macOS Mojave 10.14.6`

Problem was originally observed on CentOS 7.7 with scapy v2.4.3 using python 3.6. Tested using current github master branch (as above) on macOS to confirm.

#### How to reproduce

```
from scapy.layers.l2 import Ether
from scapy.layers.inet6 import IPv6
from scapy.layers.vrrp import VRRPv3

pkt = (Ether(src="00:00:5e:00:02:64",dst="33:33:00:00:00:12")/IPv6(src="2001:db8::1",dst="ff02::12")/VRRPv3(vrid=100,ipcount=1,addrlist=[ "2001:db8::100" ]))
```

#### Actual result

```
Traceback (most recent call last):
  File "/Users/mgsmith/src/scapy/scapy/fields.py", line 535, in h2i
    inet_aton(x)
OSError: illegal IP address string passed to inet_aton

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/mgsmith/src/scapy/scapy/base_classes.py", line 266, in __call__
    i.__init__(*args, **kargs)
  File "/Users/mgsmith/src/scapy/scapy/packet.py", line 157, in __init__
    self.fields[fname] = self.get_field(fname).any2i(self, value)
  File "/Users/mgsmith/src/scapy/scapy/fields.py", line 1412, in any2i
    return [self.field.any2i(pkt, e) for e in x]
  File "/Users/mgsmith/src/scapy/scapy/fields.py", line 1412, in <listcomp>
    return [self.field.any2i(pkt, e) for e in x]
  File "/Users/mgsmith/src/scapy/scapy/fields.py", line 562, in any2i
    return self.h2i(pkt, x)
  File "/Users/mgsmith/src/scapy/scapy/fields.py", line 537, in h2i
    x = Net(x)
  File "/Users/mgsmith/src/scapy/scapy/base_classes.py", line 108, in __init__
    self.parsed, self.netmask = self._parse_net(net)
  File "/Users/mgsmith/src/scapy/scapy/base_classes.py", line 101, in _parse_net
    tmp[0] = socket.gethostbyname(tmp[0])
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```

#### Expected result

No error, valid packet created.

#### Related resources

RFC 5798
I have a patch which adds this support which I will submit as a pull request. See https://github.com/mgsmith1000/scapy/commit/020a6695d48d16b5521bf78a4a1ef5efba14fd21.

### PR DESCRIPTION
**Pull request by mgsmith1000**
Add VRRPv3 support for IPv6 address list. The addrlist field only
supports IPv4 addresses. Make it conditional on IP as the underlayer
and add another conditional addrlist6 which is conditional on IPv6
as the underlayer.

fixes #2430

### CODE REVIEW

**Comment by mgsmith1000** [comment]
With patch applied:

**Comment by mgsmith1000** [comment]
The failures in the CI tests seem to be completely unrelated to the change I made. Could the Travis CI and AppVeyor tests be rerun for this PR to see if they pass?

File: scapy/layers/vrrp.py [hunk 1]
```diff
@@ -62,9 +62,16 @@ class VRRPv3(Packet):
         BitField("res", 0, 4),
         BitField("adv", 100, 12),
         XShortField("chksum", None),
-        # FIXME: addrlist should also allow IPv6 addresses :/
-        FieldListField("addrlist", [], IPField("", "0.0.0.0"),
-                       count_from=lambda pkt: pkt.ipcount)]
+        # use addrlist for IPv4, addrlist6 for IPv6
```
Comment (guedou) [inline scapy/layers/vrrp.py]: Can you use `addrlist` for both IPv4 and IPv6?

File: scapy/layers/vrrp.py [hunk 2]
```diff
@@ -62,9 +62,16 @@ class VRRPv3(Packet):
...
+        ConditionalField(
+            FieldListField("addrlist", [], IPField("", "0.0.0.0"),
+                           count_from=lambda pkt: pkt.ipcount),
+            lambda p: isinstance(p.underlayer, IP)),
+        ConditionalField(
+            FieldListField("addrlist6", [], IP6Field("", "::"),
+                           count_from=lambda pkt: pkt.ipcount),
+            lambda p: isinstance(p.underlayer, IPv6))
```
Comment (gpotter2) [inline scapy/layers/vrrp.py]: Could you rather use `MultipeTypeField`. This would allow to set a default value in the case that there is no underlayer.

Comment (mgsmith1000) [reply scapy/layers/vrrp.py]: @gpotter2 Yes I can. I prepared a patch to do this yesterday in order to fulfill request from @guedou to use the same field name for IPv4 and IPv6 addresses. I am testing now and will update the PR when it is complete.

File: scapy/layers/vrrp.py (hunk 1 above)
Comment (mgsmith1000) [reply scapy/layers/vrrp.py]: The PR is updated. IPv4 and IPv6 addresses both use the addrlist field now.

File: scapy/layers/vrrp.py (hunk 2 above)
Comment (mgsmith1000) [reply scapy/layers/vrrp.py]: The PR is updated and now uses MultipleTypeField for addrlist.

**Comment by mgsmith1000** [comment]
> Thanks for this PR. I made a comment.
> 
> Also, could you add unit tests such as these ones https://github.com/secdev/scapy/blob/master/test/regression.uts#L10565 It is find to use the 3 current VRRP tests and add 3 IPv6 specific ones.
> 
> Thanks.

@guedou Three new unit tests have been added for IPv6 VRRPv3 packets.

**Pull request merged**